test/lib/pwa.js

```diff
@@ -32,39 +36,114 @@ var assert = require('chai').assert;
+  var manifest;
+  var pwa;
+  var lighthouse;
+  before(done => {
+    dataFetcher.readFile(MANIFEST_DATA)
+      .then(jsonString => {
+        manifest = Manifest.fromJson(MANIFEST_URL, JSON.parse(jsonString));
+        pwa = new Pwa(MANIFEST_URL, manifest);
+        pwa.id = '123456789';
+        dataFetcher.readFile(LIGHTHOUSE_JSON_EXAMPLE)
+          .then(data => {
+            lighthouse = new Lighthouse('123456789', 'www.domain.com',
+              libLighthouse.processLighthouseJson(JSON.parse(data)));
+            done();
+          });
+      });
+  });
+
   describe('#updateIcon', () => {
+    afterEach(() => {
+      simpleMock.restore();
+    });
     it('sets iconUrl', () => {
-      return dataFetcher.readFile(MANIFEST_DATA)
-        .then(jsonString => {
-          var manifest = Manifest.fromJson(MANIFEST_URL, JSON.parse(jsonString));
-          var pwa = new Pwa(MANIFEST_URL, manifest);
-          pwa.id = '123456789';
+      // Mock libImages and bd to avoid making real calls
+      simpleMock.mock(libImages, 'fetchAndSave').resolveWith(manifest.getBestIconUrl());
+      simpleMock.mock(db, 'update').returnWith(pwa);
+      simpleMock.mock(libPwa, 'libImages').returnWith(libImages);
+      simpleMock.mock(libPwa, 'db').returnWith(db);
+
+      return libPwa.updateIcon(pwa, manifest).should.be.fulfilled.then(updatedPwa => {
+        assert.equal(libImages.fetchAndSave.callCount, 1);
+        assert.equal(libImages.fetchAndSave.lastCall.args[0],
+          'https://s1.trrsf.com/fe/zaz-morph/_img/launcher-icon.png?v2');
+        assert.equal(libImages.fetchAndSave.lastCall.args[1], '123456789.png');
+        assert.equal(db.update.callCount, 1);
+        assert.equal(updatedPwa.iconUrl,
+          'https://s1.trrsf.com/fe/zaz-morph/_img/launcher-icon.png?v2');
+      });
+    });
+  });
+
+  describe('#updateLighthouseInfo', () => {
+    afterEach(() => {
+      simpleMock.restore();
+    });
+    it('sets lighthouseScore', () => {
+      // Mock libLighthouse and bd to avoid making real calls
+      simpleMock.mock(libLighthouse, 'fetchAndSave').resolveWith(lighthouse);
+      simpleMock.mock(db, 'update').returnWith(pwa);
+      simpleMock.mock(libPwa, 'libLighthouse').returnWith(libLighthouse);
+      simpleMock.mock(libPwa, 'db').returnWith(db);
+      return libPwa.updateLighthouseInfo(pwa).should.be.fulfilled.then(updatedPwa => {
+        assert.equal(libLighthouse.fetchAndSave.callCount, 1);
+        assert.equal(libLighthouse.fetchAndSave.lastCall.args[0], '123456789');
+        assert.equal(db.update.callCount, 1);
+        assert.equal(updatedPwa.lighthouseScore, 82);
+      });
+    });
+  });
 
-          // Mock libImages and bd to avoid making real calls
-          simpleMock.mock(libImages, 'fetchAndSave').resolveWith(manifest.getBestIconUrl());
-          pwaLib.libImages = libImages;
-          simpleMock.mock(db, 'update').returnWith(pwa);
-          pwaLib.db = db;
+  describe('#save (core logic)', () => {
```

Comment:
👍 

This is all so laborious—I couldn't be bothered doing it—glad we have something now!